enterprise/server/backends/pebble_cache/pebble_cache.go

```diff
@@ -2650,9 +2654,22 @@ func (e *partitionEvictor) generateSamplesForEviction(quitChan chan struct{}) er
 			case e.samples <- sample:
 			case <-quitChan:
 				return nil
+			default:
+				// e.samples is full. Let's close the iter first, so that zombie
+				// tables can be cleaned up by pebble.
+				iter.Close()
+				shouldCreateNewIter = true
+				iter = nil
+				select {
+				case e.samples <- sample:
+				case <-quitChan:
+					return nil
+				}
```

Comment:
If the evictor is not consuming from `e.samples` faster than we're sampling, it seems like we could potentially close the iterator on every loop iteration. Does this seem like it could be an issue? Wonder if the time-based logic here might make sense after all

Reply:
We sample 10 (deletes_per_eviction) or 500 (sample_pool_size) at the same time. The reason e.samples block is when the partition size hasn't reached the threshold; Once the threshold is reached, it should consume faster.